examples/error-destructure-binding-pattern/error_destructure_binding_pattern.bal

```diff
@@ -1,36 +1,42 @@
 import ballerina/io;
 
-type SampleError error<string, map<anydata|error>>;
+type SampleErrorData record {
+    string message?;
+    error cause?;
+    string info;
+    boolean fatal;
+};
+
+type SampleError error<string, SampleErrorData>;
 
 public function main() {
     string reason;
-    map<anydata|error> detail;
+    string info;
+    boolean fatal;
     // This error-destructure binding pattern will destructure an `error` value of the type `SampleError`
-    // and assign the values to two variable references as follows:
+    // and assign the values to relevant variable references.
     // The value of the reason string in the `SampleError` will be assigned to the variable `reason`.
-    // The value of the detail mapping will be assigned to the variable `detail`.
-    error(reason, detail) = getSampleError();
+    // The values in the detail mapping will be assigned to the relevant variables.
```

Comment:
```suggestion
    // The values in the detailed mapping will be assigned to the relevant variables.
```

Reply:
Shall we change this to something like
"Fields in the error detail record will be assigned to specified variables, `info` and `fatal`.

Reply:
@praneesha 'detail' here refers to error's detail part. error has 2 parts, a reason and detail record value